### PR TITLE
style: use black and isort

### LINKED ISSUE #117: tidy up python SDK
- [x] Use GUIDs in tests
- [x] Use poetry to manage dependencies
- [x] Use flake8, isort
~~- [ ] Use pydantic for validation~~
- [x] Remove init functions. These are emulating builder patterns, which are not necessary in Python. Python supports optional and keyword arguments which obviate the need for builders.
- [x] Use pytest for testing

### PR DESCRIPTION
**Pull request by malandis**
This PR configures black and adds the isort tool. We use a modern max line length of 120 to not overly indent.

This PR also adds a  `Makefile`, which mirrors the GitHub actions. This allows developers to run the format-lint-test cycle easily.

Closes #117

### CODE REVIEW

**Review by eaddingtonwhite — COMMENTED** [review]
Looks good @malandis should we add Make as pre-req in README now? Other then that this looks gtg to me.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	@echo.
+
+.PHONY: clean
+## Remove intermediate files
+clean:
+	@rm -rf dist .mypy_cache .pytest_cache
+	@find -name "*__pycache__*" | xargs rm -rf
+
+
+.DEFAULT_GOAL := show-help
```
Comment (eaddingtonwhite) [inline Makefile:45]: Nice good idea. I usually have build be default but help makes more sense I like forced you to be explicit

**Review by cprice404 — APPROVED** [review]

**Comment by malandis** [comment]
> Looks good @malandis should we add Make as pre-req in README now? Other then that this looks gtg to me.

Thanks for the feedback. I'll add it as optional in the README.

File: Makefile (hunk 1 above)
Comment (malandis) [reply Makefile]: I think the `precommit` target is close to what `build` means. I like your idea though and added an `all` target which does the format-lint-test cycle [here](https://github.com/momentohq/client-sdk-python/pull/138/commits/237febe65a5a985358ef218620475b566b4deb7c)

Comment (malandis) [reply Makefile]: Updated contributed documentation in https://github.com/momentohq/client-sdk-python/pull/138/commits/e9f0d5cbc75525601ca5ca992cc46a226c6814c9

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -124,6 +124,9 @@ To test your changes in a python shell, just use poetry:
 poetry run pytest
 ```
 
+We've also project a `Makefile` for convenience. Install `make` then run `make help`
```
Comment (eaddingtonwhite) [inline CONTRIBUTING.md:127]:
```suggestion
We've also have a project `Makefile` for convenience. Install `make` then run `make help`
```

**Review by eaddingtonwhite — APPROVED** [review]

**Pull request merged**